runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -0,0 +1,394 @@
+import net.runelite.api.ChatPlayer;
+import net.runelite.api.ClanMember;
+import net.runelite.api.Client;
+import net.runelite.api.Friend;
+import net.runelite.api.GameState;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.api.events.PlayerMenuOptionClicked;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.chat.ChatColorType;
+import net.runelite.client.chat.ChatMessageBuilder;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.chat.QueuedMessage;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.HotkeyListener;
+import net.runelite.client.util.Text;
+import net.runelite.http.api.worlds.World;
+import net.runelite.http.api.worlds.WorldClient;
+import net.runelite.http.api.worlds.WorldResult;
+import net.runelite.http.api.worlds.WorldType;
+import org.apache.commons.lang3.ArrayUtils;
+
+@PluginDescriptor(
+	name = "World Hopper",
+	description = "Allows you to quickly hop worlds"
+)
+@Slf4j
+public class WorldHopperPlugin extends Plugin
+{
+	private static final String HOP_TO = "Hop-to";
+	private static final String KICK_OPTION = "Kick";
+	private static final ImmutableList<String> BEFORE_OPTIONS = ImmutableList.of("Add friend", "Remove friend", KICK_OPTION);
+	private static final ImmutableList<String> AFTER_OPTIONS = ImmutableList.of("Message");
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private KeyManager keyManager;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Inject
+	private ScheduledExecutorService executorService;
+
+	@Inject
+	private WorldHopperConfig config;
+
+	@Provides
+	WorldHopperConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WorldHopperConfig.class);
+	}
+
+	private final HotkeyListener previousKeyListener = new HotkeyListener(() -> config.previousKey())
+	{
+		@Override
+		public void hotkeyPressed()
+		{
+			hop(true);
+		}
+	};
+
+	private final HotkeyListener nextKeyListener = new HotkeyListener(() -> config.nextKey())
+	{
+		@Override
+		public void hotkeyPressed()
+		{
+			hop(false);
+		}
+	};
+
+	private Future<?> worldResultFuture;
+	private WorldResult worldResult;
+
+	private int worldToHopTo = -1;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(previousKeyListener);
+		keyManager.registerKeyListener(nextKeyListener);
+
+		worldResultFuture = executorService.submit(() ->
+		{
+			try
+			{
+				WorldResult worldResult = new WorldClient().lookupWorlds();
+				worldResult.getWorlds().sort(Comparator.comparingInt(World::getId));
+				this.worldResult = worldResult;
+			}
+			catch (IOException ex)
+			{
+				log.error(null, ex);
```

Comment:
I would add some decription to this. Also, make this just warning, error usually means app cannot continue working, this will just make the plugin not work correctly.